lib/copydir.c

```diff
@@ -606,18 +606,6 @@ static int copy_symlink (const struct path_info *src, const struct path_info *ds
 		return -1;
 	}
 
-	/* If src was a link to an entry of the src_orig directory itself,
```

Comment:
First of all, I really hate this code, and its use of the global src_orig.

But, I do not see the problem with the code that is there.  Is there a security issue in how it's done?  Any link source will also have been copied, so functionally this will be fine.

And, most importantly, it risks breaking a lot of installations for no seeming benefit.  So I'm -1 on this, sorry.

Reply:
Tobias had a reproducer.  It's not a vulverability (at first glance), but it's not nice either:

```sh
ln -s /etc/skel2 /etc/skel/link
useradd -m user
ls -l /home/user
```

<https://github.com/shadow-maint/shadow/issues/933#issue-2110878747>

Reply:
And yes, fixing that might break existing stuff.  :|